aries-test-harness/features/0793-peer-did.feature

```diff
@@ -0,0 +1,96 @@
+      # This will rely on the wallet type being used on the ./manage start command.
+      # For example the EXTRA_ARGS environment variable or passing an ACA-Py config file like issuer_anoncreds.yaml with the AGENT_CONFIG_FILE environment variable.
+      # Examples:
+      #    | peer_did_method | start_parameters                                               |
+      #    | did:peer:1      | {"flags":["emit-did-peer-1"]} |
+      #    | did:peer:2      | {"flags":["emit-did-peer-2"]} |
+      #    | did:peer:3      | {"flags":["emit-did-peer-3"]} |
+      #    | did:peer:4      | {"flags":["emit-did-peer-4"]} |
+
+      # These Sceanrio Examples can be used if we run the these did:peer tests in the same runset as the unqualified did:peer tests.
+      # This will restart acapy with a different wallet type.
+      Examples:
+         | peer_did_method | start_parameters                                               |
+         | did:peer:1      | {"wallet-type":"askar-anoncreds", "flags":["emit-did-peer-1"]} |
+         | did:peer:2      | {"wallet-type":"askar-anoncreds", "flags":["emit-did-peer-2"]} |
+         | did:peer:3      | {"wallet-type":"askar-anoncreds", "flags":["emit-did-peer-3"]} |
+         | did:peer:4      | {"wallet-type":"askar-anoncreds", "flags":["emit-did-peer-4"]} |
+
+
+   @T003-RFC0793 @critical @AcceptanceTest
+   Scenario Outline: Establish a connection with DID Exchange between two agents utilizing qualified did:peer DIDs with responder set to send a different Peer DID method
+      Given we have "2" agents
+         | name | role      | start_parameters             |
+         | Acme | requester | <start_parameters>           |
+         | Bob  | responder | <responder_start_parameters> |
+      When "Bob" sends an invitation to "Acme" with <peer_did_method>
+         | use_public_did |
+         | True           |
+      And "Acme" receives the invitation
+      And "Acme" sends the request to "Bob"
+      And "Bob" receives the request
+      And "Bob" sends a response to "Acme"
+      And "Acme" receives the response
+      And "Acme" sends complete to "Bob"
+      Then "Acme" and "Bob" have a connection
+
+      Examples:
+         | peer_did_method | start_parameters                                               | responder_start_parameters                                     |
+         | did:peer:1      | {"wallet-type":"askar-anoncreds", "flags":["emit-did-peer-1"]} | {"wallet-type":"askar-anoncreds", "flags":["emit-did-peer-3"]} |
+         | did:peer:2      | {"wallet-type":"askar-anoncreds", "flags":["emit-did-peer-2"]} | {"wallet-type":"askar-anoncreds", "flags":["emit-did-peer-4"]} |
+         | did:peer:3      | {"wallet-type":"askar-anoncreds", "flags":["emit-did-peer-3"]} | {"wallet-type":"askar-anoncreds", "flags":["emit-did-peer-1"]} |
+         | did:peer:4      | {"wallet-type":"askar-anoncreds", "flags":["emit-did-peer-4"]} | {"wallet-type":"askar-anoncreds", "flags":["emit-did-peer-2"]} |
```

Comment:
Actually, comments: I'm not sure it makes sense for use to include tests for did:peer:3. Also, ACA-Py has not implemented an `emit-did-peer-1` mode.

Reply:
I've added them in for when they are implemented these tests should then automatically pass. However, if we prefer, we can just comment them out until or if we support those modes. 
I like a test first approach, where we have failing tests then the software catches up to the tests to make them pass. I can understand that we may not want that here, showing a higher fail rate for unimplemented features.

Reply:
@swcurran Would you prefer currently unsupported DID Peer Methods to be commented out or leave them failing? I'm inclined to think you would want them commented out.

Reply:
We are unlikely to implement support for emitting a did:peer:1 or 3

Reply:
Will any other Aries Framework implement those? The RFC mentions them so that is why I put them in.

Reply:
AFJ implements 1 but it's actually not possible to exchange just did:peer:3s. You emit 2 and then derive a 3.